archaius2-api/src/main/java/com/netflix/archaius/api/Property.java

```diff
@@ -77,16 +85,20 @@ default void removeListener(PropertyListener<T> listener) {}
      * since the notification only applies to the state of the chained property
      * up until this point. Changes to subsequent Property objects returned from {@link Property#orElse} 
      * or {@link Property#map(Function)} will not trigger calls to this consumer.
+
      * @param consumer
-     * @return This property object
+     * @return Subscription that may be unsubscribed to no longer get change notifications
      */
-    default void onChange(Consumer<T> consumer) {
-        addListener(new PropertyListener<T>() {
+    default Subscription onChange(Consumer<T> consumer) {
```

Comment:
nit: semantically 'Subscription' isn't a very obvious result from the 'onChange' method; would 'subscribe' work as the method name here?  What is the terminology in RexJava?